controllers/edgedeployment_controller.go

```diff
@@ -184,8 +185,9 @@ func (r *EdgeDeploymentReconciler) addDeploymentsToDevices(ctx context.Context,
 	for _, edgeDevice := range edgeDevices {
 		if !hasDeployment(edgeDevice, name) {
 			deploymentStatus := managementv1alpha1.Deployment{Name: name, Phase: managementv1alpha1.Deploying}
+			patch := client.MergeFrom(edgeDevice.DeepCopy())
 			edgeDevice.Status.Deployments = append(edgeDevice.Status.Deployments, deploymentStatus)
-			err := r.EdgeDeviceRepository.UpdateStatus(ctx, &edgeDevice)
```

Comment:
`UpdateStatus` can be removed from the `EdgeDeviceRepository`; let's not leave too many options for updating status.